framework/py/flwr/common/record/arrayrecord.py

```diff
@@ -365,6 +367,66 @@ def count_bytes(self) -> int:
 
         return num_bytes
 
+    @property
+    def children(self) -> dict[str, InflatableObject]:
+        """Return a dictionary of Arrays with their Object IDs as keys."""
+        return {arr.object_id: arr for arr in self.values()}
+
+    def deflate(self) -> bytes:
+        """Deflate the ArrayRecord."""
+        # array_name: array_object_id mapping
+        array_refs: dict[str, str] = {}
+
+        for array_name, array in self.items():
+            array_refs[array_name] = array.object_id
+
+        # Serialize references dict
+        object_body = json.dumps(array_refs).encode("utf-8")
+        return add_header_to_object_body(object_body=object_body, cls=self)
+
+    @classmethod
+    def inflate(
+        cls, object_content: bytes, children: dict[str, InflatableObject] | None = None
+    ) -> ArrayRecord:
+        """Inflate an ArrayRecord from bytes.
+
+        Parameters
+        ----------
+        object_content : bytes
+            The deflated object content of the ArrayRecord.
+        children : Optional[dict[str, InflatableObject]] (default: None)
+            Dictionary of children InflatableObjects mapped to their Object IDs.
+            These children enable the full inflation of the ArrayRecord.
+
+        Returns
+        -------
+        ArrayRecord
+            The inflated ArrayRecord.
+        """
+        if children is None:
+            children = {}
+
+        # Inflate mapping of array_names (keys in the ArrayRecord) to Arrays' object IDs
+        obj_body = get_object_body(object_content, cls)
+        array_refs: dict[str, str] = json.loads(obj_body.decode(encoding="utf-8"))
+
+        if len(array_refs) != len(children):
+            raise ValueError(
+                "Unexpected number of `children`. "
+                f"Expected {len(array_refs)} but got {len(children)}."
+            )
```

Comment:
Isn't this incorrect? If an `ArrayRecord` includes the same `Array` twice under a different key, then we have more elements in the `ArrayRecord` dict than we have in the `children` dict.